src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -200,10 +208,59 @@ internal int SerializeLogRecord(LogRecord logRecord)
             var name = logRecord.CategoryName;
 
             // If user configured explicit TableName, use it.
-            if (this.m_tableMappings == null || !this.m_tableMappings.TryGetValue(name, out var eventName))
+            if (this.m_tableMappings == null || (!this.m_tableMappings.TryGetValue(name, out var eventName) && !this.shouldPassThruTableMappings))
             {
                 eventName = this.m_defaultEventName;
             }
+            else if (this.shouldPassThruTableMappings && eventName == null)
+            {
+                char[] tempArr = new char[name.Length];
+                int readIdx = 0;
+                int writeIdx = 0;
+                while (readIdx < name.Length)
+                {
+                    if (readIdx == 0)
+                    {
+                        if (name[readIdx] >= 'A' && name[readIdx] <= 'Z')
+                        {
+                            tempArr[writeIdx] = name[readIdx];
+                            ++writeIdx;
+                        }
+                        else if (name[readIdx] >= 'a' && name[readIdx] <= 'z')
+                        {
+                            // If the first character in the resulting string is lower -case ALPHA,
+                            // it will be converted to the corresponding upper-case.
+                            tempArr[writeIdx] = (char)(name[readIdx] - 32);
+                            ++writeIdx;
+                        }
+
+                        // Not a valid name - Part B name should follow PascalCase naming convention.
+                        else
+                        {
+                            break;
+                        }
+                    }
+
+                    // The category name must match "^[A-Z][a-zA-Z0-9]*$"; any character that is not allowed will be removed.
```

Comment:
```suggestion
                    // The category name must match "^[A-Z][a-zA-Z0-9]{0,31}$"; any character that is not allowed will be removed.
```
Is this more accurate?